lib/internal/crypto/random.js

```diff
@@ -119,6 +120,51 @@ function randomFill(buf, offset, size, cb) {
   _randomBytes(buf, offset, size, wrap);
 }
 
+// Largest integer we can read from a buffer.
+// e.g.: Buffer.from("ff".repeat(6), "hex").readUIntBE(0, 6);
+const RAND_MAX = 281474976710655;
+
+function randomInt(min, max, cb) {
+  if (typeof cb !== 'function') {
+    throw new ERR_INVALID_CALLBACK(cb);
+  }
+  if (!Number.isInteger(min)) {
+    throw new ERR_INVALID_ARG_TYPE('min', 'integer', min);
+  }
+  if (!Number.isInteger(max)) {
+    throw new ERR_INVALID_ARG_TYPE('max', 'integer', min);
+  }
+  if (!(max >= min)) {
+    throw new ERR_OUT_OF_RANGE('max', `>= ${min}`, max);
+  }
+  if (!(max <= RAND_MAX)) {
+    throw new ERR_OUT_OF_RANGE('max', `<= ${RAND_MAX}`, max);
+  }
+
+  // First we generate a random int between [0..bound)
+  const bound = max - min;
+  const range = bound - 1;
+  const excess = RAND_MAX % range;
+  const randLimit = RAND_MAX - excess;
+
+  const pickAttempt = () => {
+    randomBytes(6, (err, bytes) => {
+      if (err) return cb(err);
+      const x = bytes.readUIntBE(0, 6);
+      // If x > (maxVal - (maxVal % range)), we will get "modulo bias"
+      if (x > randLimit) {
+        // Try again
+        return pickAttempt();
+      }
+
+      const n = (x % range) + min;
+      cb(null, n + min);
+    });
+  };
+
+  pickAttempt();
```

Comment:
I suspect that this function could be much faster when implemented in C++ using OpenSSL functions. (Calling `randomBytes` is expensive.)

Reply:
I think it’s safe to assume that we’ll use only one iteration each time here in most real-world scenarios (i.e. when the range isn’t specifically chose to increase the likelihood of having to pick again, and even there the maximum expected value for the number of attempts is just 2)

Reply:
Yes, my point is that a C++ implementation could avoid creating a buffer, converting it to a JS number etc. (But I don't know how large the performance impact actually is.)

Reply:
We could try to measure it, but I don’t think it’s worth it, tbh. :)